go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -118,8 +118,13 @@ func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, ignoreLearn bool) error {
 	if err != nil {
 		return err
 	}
-	namespaceExternalGw := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
-	namespaceVTEP := namespace.GetAnnotations()[hotypes.HybridOverlayVTEP]
+
+	_ = n.copyNamespaceAnnotationsToPod(namespace, pod, false)
+
+	namespaceExternalGw := pod.GetAnnotations()[hotypes.HybridOverlayExternalGw]
+	namespaceVTEP := pod.GetAnnotations()[hotypes.HybridOverlayVTEP]
+	ignoreLearn := pod.GetAnnotations()[hotypes.HybridOverlayIgnoreLearn] != ""
```

Comment:
i dont think we need an annotation to ignoreLearn, i think any addOrUpdatePod can set ignoreLearn to true

Reply:
reasons are:

1. pod add - there are no flows for this pod, so we can safely ignore learn for this pod
2. pod update - we only should update the pod if pod IPs changed, in which case we need to update the flow anyway, so its safe to ignore the learn here
3. namespace update - new vteps or gw and we need to ignore learn

So in every possible k8s event for a pod, its safe to ignore learn for the pod. It's not OK to always ignore learn, we need to always learn unless the update is being triggered due to a k8s event and not a periodic sync.

Reply:
@trozet even the `addOrUpdatePod` calls that are made when the namespace is updated?